plugins/woocommerce/src/Admin/API/Reports/Customers/DataStore.php

```diff
@@ -835,6 +861,11 @@ public static function delete_customer( $customer_id ) {
 	public static function delete_customer_by_user_id( $user_id ) {
 		global $wpdb;
 
+		if ( (int) $user_id < 1 || doing_action( 'wp_uninitialize_site' ) ) {
```

Comment:
I'm wondering why this had to be added. In which scenario would this be called with user_id < 1? Maybe it's worth adding a comment?

EDIT: I see you only moved this from the scheduler. Feel free to leave it like this if you think it's OK.

Reply:
A user ID of `0` would be an unauthenticated request, or some error up the chain that resulted in user ID `0`. Whatever the case, proceeding with the `DELETE` would be unnecessary, so the early return seems most robust.